### ISSUE TITLE
selector-max-id throws error when using :global in webpack css-loader

### ISSUE THREAD

**Issue by inferusvv** [stylelint/stylelint#2927]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

I have webpack css-loader and use `:global` in my SASS file to make class global and don't transform it. I am getting `selector-max-id` lint error

> Which rule, if any, is this issue related to?

`selector-max-id`

> What CSS is needed to reproduce this issue?

e.g.

```css
:global {
  #app {
    height: 100%;
  }
}
```

> Which version of stylelint are you using?

`8.1.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint \"src/**/*.{css,less,scss,sss}\"`

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
10:3  ✖  Expected ":global #app" to have no more than 0 id selectors   selector-max-id
```

**Comment by hudochenkov** [comment]
What stylelint configuration is needed to reproduce this issue?

**Comment by inferusvv** [comment]
@hudochenkov I think it's enough just enable `selector-max-id` rule. But here is my full `.stylelint.rc` file (from react-starter-kit by kriasoft)

```
module.exports = {
  // The standard config based on a handful of CSS style guides
  // https://github.com/stylelint/stylelint-config-standard
  // extends: 'stylelint-config-standard',
  extends: 'stylelint-config-sass-guidelines',

  plugins: [
    // stylelint plugin to sort CSS rules content with specified order
    // https://github.com/hudochenkov/stylelint-order
    'stylelint-order',
    'stylelint-scss'
  ],

  rules: {
```
    'property-no-unknown': [
      true,
      {
        ignoreProperties: [
          // CSS Modules composition
          // https://github.com/css-modules/css-modules#composition
          'composes',
        ],
      },
    ],

    'selector-pseudo-class-no-unknown': [
      true,
      {
        ignorePseudoClasses: [
          // CSS Modules :global scope
          // https://github.com/css-modules/css-modules#exceptions
          'global',
        ],
      },
    ],

    // Opinionated rule, you can disable it if you want
    'string-quotes': 'single',

    // https://github.com/hudochenkov/stylelint-order/blob/master/rules/order/README.md
    'order/order': [
      'custom-properties',
      'dollar-variables',
      'declarations',
      'at-rules',
      'rules',
    ],

    // https://github.com/hudochenkov/stylelint-order/blob/master/rules/properties-order/README.md
    'order/properties-order': [],
  },
};

**Comment by hudochenkov** [comment]
Config you extend has `selector-max-id` enabled. https://github.com/bjankord/stylelint-config-sass-guidelines/blob/81323826dffaaa6b39c522d4698f21f365903a1c/index.js#L71

If you need to use id selector you can [disable this rule](https://stylelint.io/user-guide/configuration/#turning-rules-off-from-within-your-css) for part of your style sheet.

**Comment by inferusvv** [comment]
@hudochenkov Yes, I disabled it. But is it ok? There is only one id, but rule thinks that 2

**Comment by jeddy3** [comment]
@inferusvv Thanks for the using the template.

> But is it ok? There is only one id, but rule thinks that 2

Yes, it's ok. If you look closely at the error you'll see that the rules think _there are more than 0 id selectors_:

```
10:3  ✖  Expected ":global #app" to have no more than 0 id selectors   selector-max-id
```

Which there are, as the following:

```css
:global {
  #app {
    height: 100%;
  }
}

```

Contains _one_ id selector `#app`.
